src/extra_files.c

```diff
@@ -63,15 +56,34 @@ static inline int bsearch_helper(const void *A, const void *B)
 
 static int record_filename(const char *name, const struct stat *stat __attribute__((unused)), int type, struct FTW *ftw __attribute__((unused)))
 {
-	for (size_t i = 0; i < sizeof(skip_dirs) / sizeof(skip_dirs[0]); i++) {
-		if (strncmp(name, skip_dirs[i].name, skip_dirs[i].len) == 0) {
+	/* Name as it would appear in manifest, f.i. /usr */
+	const char *relname = name + path_prefix_len - 1;
+
+	if (!relname[0] || !strcmp(relname, "/")) {
+		/* Do not record root while descending into it. */
+		return 0;
+	}
+
+	if (path_whitelist) {
```

Comment:
path_whitelist will now always be set, so we can remove this conditional.

Reply:
It's not explicitly documented, but walk_tree() could be called with a NULL regex. It's not used as a general purpose utility function now, but it could be one day, so I'd prefer to not make assumptions about where the regex comes from and whether it's always set.